lib/spandx/core/http.rb

```diff
@@ -3,17 +3,24 @@
 module Spandx
   module Core
     class Http
-      attr_reader :driver
+      attr_reader :driver, :retries
 
-      def initialize(driver: Http.default_driver)
+      def initialize(driver: Http.default_driver, retries: 3)
         @driver = driver
+        @retries = retries
+        @circuits = Hash.new { |hash, key| hash[key] = Circuit.new }
```

Comment:
why do we need `Circuit.new` here ? 

later we do 

```ruby 
  @circuits[host] = false
```

Reply:
This is a quick and dirty implementation of a [circuit breaker](https://www.martinfowler.com/bliki/CircuitBreaker.html). As soon as the scan detects a host that we cannot reach the circuit is opened for that host and this class wont waste time trying to connect to the host over and over for each dependency. I came across this while I scanned my full development directory.